### PR TITLE
Jx ss2 reproducibility 558

### PR DESCRIPTION
**Pull request by jishuxu**
In this PR
1) T-test between SCs samples and bulk samples for both base and updated pipeline then compare the signature genes between two pipeline output. Highlight  inconsistent signature genes, such as significant in base pipeline but not significant in updated pipeline. Visualize this result in scatter plot, such as x-axis is mean(Bulk) - mean(SCs) from base pipeline and Y-axis is mean(Bulk) - mean(SCs) from updated pipeline and color theme represents the significant status between pipelines. 
2) also summary these inconsistent significant genes by their biotypes. visualize this in barplot 
3) pull common function into a analysis_functions.R file
4) example output gs://broad-dsde-mint-dev-cromwell-execution/cromwell-executions/RunDataMatrixReproducibility/dd09e36f-5219-4852-ac87-b0c47359e48d/call-DataMatrixReproducibility/STAR_HISAT2_TPM_rep.tar.gz

### CODE REVIEW

**Comment by jishuxu** [comment]
graphic output gs://broad-dsde-mint-dev-cromwell-execution/cromwell-executions/RunDataMatrixReproducibility/dd09e36f-5219-4852-ac87-b0c47359e48d/call-DataMatrixReproducibility/STAR_HISAT2_TPM_rep.tar.gz

**Comment by jishuxu** [comment]
output gs://broad-dsde-mint-dev-cromwell-execution/cromwell-executions/RunDataMatrixReproducibility/dd09e36f-5219-4852-ac87-b0c47359e48d/call-DataMatrixReproducibility/STAR_HISAT2_TPM_rep.tar.gz

File: benchmarking/smartseq2/R/analysis_functions.R [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  dtable <-
+    data.frame(
+      'test' = 'K-S',
+      'D-stats' = round(ks$statistic, 4),
+      'P-value' = convert2star(ks$p.value)
+    )
+  dtable.p <-
+    ggtexttable(dtable,
+                row = NULL,
+                theme = ttheme('mBlue', base_size = 18))
```
Comment (meganshand) [inline benchmarking/smartseq2/R/analysis_functions.R]: Shouldn't this be `ttheme -> theme`? How did this work?

File: benchmarking/smartseq2/R/analysis_functions.R [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+  return(data.frame(
+    'pvalue' = pval,
+    'cor' = cval,
+    'sample' = colnames(x)
+  ))
+}
+# calculate correlation matrix btw matrix data
+CorrDataMatrix <- function(mat1, mat2, islog2) {
+  # match gene ID
+  if (islog2 == 1) {
```
Comment (meganshand) [inline benchmarking/smartseq2/R/analysis_functions.R]: Can you make the changes from the comments about this function in #94 here as well?

File: benchmarking/smartseq2/matrix_reproducibility/data_matrix_reproducibility.R [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    x = 'Base',
+    y = 'Updated',
+    color = "groups",
+    size = 3,
+    alpha = 0.6,
+    palette = "jco",
+    ggtheme = theme_minimal(),
+    legend.title = "t-test SC vs Bulk",
+    xlab = "Base Bulk -SC",
+    ylab = "Updated Bulk - SC"
```
Comment (meganshand) [inline benchmarking/smartseq2/matrix_reproducibility/data_matrix_reproducibility.R]: Can you say what the unit for the x and y axis are?

File: benchmarking/smartseq2/matrix_reproducibility/data_matrix_reproducibility.R [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+pb <- ggbarplot(
+  mtb,
+  x = 'Var2',
+  y = 'value',
+  color = "Var1",
+  fill = "Var1",
+  # change fill color by FC
+  x.text.angle = 90,
+  # Rotate vertically x axis texts
+  xlab = "",
```
Comment (meganshand) [inline benchmarking/smartseq2/matrix_reproducibility/data_matrix_reproducibility.R]: Can you say what the x-axis is here.

File: benchmarking/smartseq2/R/analysis_functions.R (hunk 1 above)
Comment (jishuxu) [reply benchmarking/smartseq2/R/analysis_functions.R]: it stands for `table theme` so should be `ttheme()`

File: benchmarking/smartseq2/R/analysis_functions.R (hunk 2 above)
Comment (jishuxu) [reply benchmarking/smartseq2/R/analysis_functions.R]: yes, I am changing  this for #94.

File: benchmarking/smartseq2/matrix_reproducibility/data_matrix_reproducibility.R (hunk 3 above)
Comment (jishuxu) [reply benchmarking/smartseq2/matrix_reproducibility/data_matrix_reproducibility.R]: yes, will add `fold changes` in labels

File: benchmarking/smartseq2/R/analysis_functions.R (hunk 1 above)
Comment (meganshand) [reply benchmarking/smartseq2/R/analysis_functions.R]: Ah ok. This reminds me, could you put the loading library lines in this file? That way you don't need to load the libraries ( and know which ones to load) ahead of time if you're using these functions?

**Comment by jishuxu** [comment]
@meganshand  I just updated `analysis_functions.R` and main R scripts

**Pull request merged**